packages/core/src/transport/transport.ts

```diff
@@ -1,11 +1,45 @@
 import { display } from '../tools/display'
 import { Context } from '../tools/context'
 import { addEventListener, DOM_EVENT, jsonStringify, noop, objectValues } from '../tools/utils'
-import { monitor, addErrorToMonitoringBatch } from '../domain/internalMonitoring'
+import { monitor, addErrorToMonitoringBatch, addMonitoringMessage } from '../domain/internalMonitoring'
 
 // https://en.wikipedia.org/wiki/UTF-8
 const HAS_MULTI_BYTES_CHARACTERS = /[^\u0000-\u007F]/
 
+/* eslint-disable camelcase */
+type ErrorCtx =
+  | {
+      err_msg?: string
+      err_stack?: string
+    }
+  | undefined
+
+type TransportErrorCtx = {
+  beacon: Record<string, string | number | boolean>
+  event?: Record<string, string | number | boolean>
+  req?: Record<string, string | number | boolean>
+  error_ctx?: ErrorCtx
+  on_line: boolean
+}
+/* eslint-enable camelcase */
+
+const getTransportErrorEventDetails = (evt: ProgressEvent) => {
+  const req = evt?.currentTarget as XMLHttpRequest
+  return {
+    event: {
+      type: evt?.type,
+      is_trusted: evt?.isTrusted,
+      total: evt?.total,
+      loaded: evt?.loaded,
+    },
+    req: {
+      status: req?.status,
+      ready_state: req?.readyState,
+      timeout: req?.timeout,
+    },
```

Comment:
I don't think all those `?.` are needed. If they are, why not returning early if there is no `evt` or `req`?

FYI: `foo?.bar` is transpiled to `foo === null || foo === void 0 ? void 0 : foo.bar;`, so adding unnecessary operators like this may produce a bit of extra code in the final bundle.